awx/ui/src/screens/Setting/shared/SharedFields.js

```diff
@@ -229,6 +230,35 @@ EncryptedField.propTypes = {
   config: shape({}).isRequired,
 };
 
+const ExecutionEnvField = ({ name, config, isRequired = false }) => {
+  const [field, meta, helpers] = useField({ name });
+  return config ? (
+    <SettingGroup
+      defaultValue={config.default ?? ''}
+      fieldId={name}
+      helperTextInvalid={meta.error}
+      isRequired={isRequired}
+      label={config.label}
+      popoverContent={config.help_text}
+      isDisabled={field.value === null}
+      onRevertCallback={() => helpers.setValue(config.default)}
+    >
+      <ExecutionEnvironmentLookup
+        value={field.value}
+        onChange={(value) => {
+          helpers.setValue(value, false);
+        }}
+        overrideLabel
+        fieldName={name}
```

Comment:
hello @kialam, good job. 

We need to keep those 2 props.

```js
                  popoverContent={t`The Execution Environment to be used when one has not been configured for a job template.`}
                  isGlobalDefaultEnvironment
```

Reply:
I see the changes that you added about `overrideLabel` 👍 and relying on the data returned by the API.